plugins/coverimage.koplugin/main.lua

```diff
@@ -270,7 +267,20 @@ function CoverImage:addToMainMenu(menu_items)
                             return self.cover_image_background == "gray"
                         end,
                         callback = function()
-                            self.cover_image_background = self.cover_image_background == "gray" and "none" or "gray"
+                            self.cover_image_background = "gray"
+                            G_reader_settings:saveSetting("cover_image_background", self.cover_image_background)
+                            if self.enabled then
+                                self:createCoverImage(self.ui.doc_settings)
+                            end
+                        end
+                    },
+                    {
+                        text = _("Stretch to fit screen"),
+                        checked_func = function()
+                            return self.cover_image_background == "none"
```

Comment:
Is their a 5th alternative behaviour when `self.cover_image_background` is `nil` ?
If yes, it would be nice it appears as the first option "Save cover as-is".
If no, one of these 4 should handle `self.cover_image_background == nil` in the checked_func.

Reply:
We should not have a 5th alternative, because actually we do not stretch the image to fullscreen, we save it as is. The firmware of the device can take this and scale, stretch or do whatever the device does. 
So actually the 4th menu entry should be named better, something the way: "Save image as is and let the firmware deal with stretching" (in better word of course).

`self.cover_image_background` cannot be `nil` because we initialize it in line 43 with "black" if it is nil.

Reply:
Maybe we could name it "Save the original image" and make a help_text with the info "The firmware can stretch the image to fit the screen"

Reply:
```
white
gray
black
stretch
as-is
```
That makes 5. Which one would you remove?

(Or do you mean you don't really need stretch ? And what I thought was better named stetch was actually just doing scaling to screen size, preservering aspect ratio? So, as-is just scaling to screen size, keep aspect ratio - and we can actually keep it doing that while still naming it as-is, for simplicity).

Let's keep using "scale" when the aspect ratio is preserved (and "stretch" only when it is not).

Reply:
We have:
black -> scale and add black background
white -> scale and add white background
gray -> scale and add gray background
as-is  -> just save 

Until now, the stretch option is not implemented in the plugin, because it is not needed. The firmware takes an image and stretches it to fully fit the screen (which does not work very good on Poke2, but works on Tolino).

Of course, we can add a fifth option `stretch`, if you think that's necessary.

I am not really satisfied with the naming of the menu options, wouldn't it  be better to name the thing: `Size and background` and the submenus like
```
[ ] Original size
[ ] Scale, black background
[ ] Scale, white background
[ ] Scale, gray background
```

Reply:
OK, I'm fine with that proposal (no need for stretch if it's the default behaviour on some FW, and otherwise the least interesting option as the others are better).

Reply:
That looks fine to me. :+1: